src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilderResource.java

```diff
@@ -0,0 +1,162 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.coremod.entity.ai.item.handling.ItemStorage;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Information about a resource.
+ * - How many are needed to finish the build
+ * - How many are available to the builder
+ * - How many are in the player's inventory (client side only)
+ */
+public class BuildingBuilderResource extends ItemStorage
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2160)